src/robot/parsing/model/statements.py

```diff
@@ -355,42 +577,152 @@ def name(self):
 class Setup(Fixture):
     type = Token.SETUP
 
+    @classmethod
+    def from_params(cls, keyword, args=None, indent=FOUR_SPACES, separator=FOUR_SPACES, eol=EOL):
+        tokens = [Token(Token.SEPARATOR, indent), Token(Token.SETUP, '[Setup]')]
+        if keyword:
+            tokens.append(Token(Token.SEPARATOR, separator))
+            tokens.append(Token(Token.NAME, keyword))
+        if args is not None:
+            for arg in args:
+                tokens.append(Token(Token.SEPARATOR, separator))
+                tokens.append(Token(Token.ARGUMENT, arg))
+        tokens.append(Token(Token.EOL, eol))
+        return cls(tokens)
+
 
 @Statement.register
 class Teardown(Fixture):
     type = Token.TEARDOWN
 
+    @classmethod
+    def from_params(cls, keyword, args=None, indent=FOUR_SPACES, separator=FOUR_SPACES, eol=EOL):
+        tokens = [Token(Token.SEPARATOR, indent), Token(Token.TEARDOWN, '[Teardown]')]
+        if keyword:
+            tokens.append(Token(Token.SEPARATOR, separator))
+            tokens.append(Token(Token.NAME, keyword))
+        if args is not None:
+            for arg in args:
+                tokens.append(Token(Token.SEPARATOR, separator))
+                tokens.append(Token(Token.ARGUMENT, arg))
+        tokens.append(Token(Token.EOL, eol))
+        return cls(tokens)
+
 
 @Statement.register
 class Tags(MultiValue):
     type = Token.TAGS
 
+    @classmethod
+    def from_params(cls, tags, indent=FOUR_SPACES, separator=FOUR_SPACES, eol=EOL):
+        tokens = [
+            Token(Token.SEPARATOR, indent),
+            Token(Token.TAGS, '[Tags]')
+        ]
+
+        for tag in tags:
+            tokens.append(Token(Token.SEPARATOR, separator))
+            tokens.append(Token(Token.ARGUMENT, tag))
+        tokens.append(Token(Token.EOL, eol))
+        return cls(tokens)
+
 
 @Statement.register
 class Template(SingleValue):
     type = Token.TEMPLATE
 
+    @classmethod
+    def from_params(cls, keyword=None, indent=FOUR_SPACES, separator=FOUR_SPACES, eol=EOL):
+        tokens = [
+            Token(Token.SEPARATOR, indent),
+            Token(Token.TEMPLATE, '[Template]')
+        ]
+        if keyword:
+            tokens.append(Token(Token.SEPARATOR, separator))
+            tokens.append(Token(Token.NAME, keyword))
```

Comment:
It's rarely used and probably not known (overriding suite settings by empty values) so we ought to support only non empty values in `from_params`. In case someone needs empty value there is always `from_tokens` method.